### PR TITLE
delete feature

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
https://github.com/pingcap/tispark/issues/2235

### CODE REVIEW

**Comment by shiyuhang0** [comment]
@xuanyu66 @birdstorm PTAL

File: README.md [hunk 1]
```diff
@@ -114,29 +114,26 @@ spark.sql.catalog.tidb_catalog.pd.addresses  ${your_pd_adress}
 ```
 ./bin/spark-shell --jars /wherever-it-is/tispark-${name_with_version}.jar
 ```
-
-### Use TiSpark
-For TiSpark version 2.4.x:
+### Get TiDB Version
```
Comment (xuanyu66) [inline README.md:117]: image
It’s TiSpark's version

File: README.md [hunk 2]
```diff
@@ -114,29 +114,26 @@ spark.sql.catalog.tidb_catalog.pd.addresses  ${your_pd_adress}
...
 
 ```
-spark.sql("use ${database}")
-spark.sql("select count(*) from ${table}").show
+spark.sql("select ti_version()").show
```
Comment (xuanyu66) [inline README.md:120]: use collect, not show

File: README.md [hunk 3]
```diff
@@ -164,6 +161,16 @@ customer.write
 
 See [here](./docs/datasource_api_userguide.md) for more details.
 
+### Delete with TiSpark
```
Comment (xuanyu66) [inline README.md:164]: Note with only support with version 2.6.x

File: core/src/main/scala/com/pingcap/tispark/utils/TwoPhaseCommitHepler.scala [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+ */
+case class TwoPhaseCommitHepler(startTs: Long, options: TiDBOptions) extends AutoCloseable {
+
+  // 2PC relies too much on TiDBOptions. However, TiDBOptions requires db information which 2PC needn't.
+  // So we just fake db information temporarily,
+  // TODO change TIDBOptions to TwoPhaseCommitOption
+  def this(startTs: Long) {
+    this(
+      startTs,
+      new TiDBOptions(
```
Comment (xuanyu66) [inline core/src/main/scala/com/pingcap/tispark/utils/TwoPhaseCommitHepler.scala:43]: Extract these options into a class

**Review by xuanyu66 — CHANGES_REQUESTED** [review]

File: core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala [hunk 5]
```diff
@@ -378,6 +378,15 @@ object TiAuthorization {
         f"SELECT command denied to user ${tiAuth.get.user} for table $database.$table")
   }
 
+  def authorizeForDelete(
```
[the remark this answers is not in the document]
Comment (xuanyu66) [reply core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala:381]: It's better to experiment first on TiDB or MySql

[the remark this answers is not in the document]
Comment (shiyuhang0) [reply core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala:381]: SELECT privilege has been checked in TiCatalog.loadTable.

[the remark this answers is not in the document]
Comment (shiyuhang0) [reply core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala:381]: See design doc for detail. https://github.com/pingcap/tispark/blob/41cf51cf6012b054e4e3a121540655d306b4adb3/docs/design/2022-03-21-delete-support.md#auth-design

**Review by birdstorm — APPROVED** [review]
LGTM

**Review by xuanyu66 — CHANGES_REQUESTED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Comment by xuanyu66** [comment]
LGTM

**Review by birdstorm — APPROVED** [review]
LGTM

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**